code/game/objects/items/his_grace.dm

```diff
@@ -0,0 +1,262 @@
+	step_to(src, L)
+	if(Adjacent(L))
+		if(!L.stat)
+			L.visible_message("<span class='warning'>[src] lunges at [L]!</span>", "<span class='his_grace big bold'>[src] lunges at you!</span>")
+			do_attack_animation(L, null, src)
+			playsound(L, 'sound/weapons/smash.ogg', 50, TRUE)
+			playsound(L, 'sound/misc/desceration-01.ogg', 50, TRUE)
+			L.adjustBruteLoss(force)
+			adjust_bloodthirst(-5) //Don't stop attacking they're right there!
+		else
+			consume(L)
+
+/obj/item/his_grace/proc/awaken(mob/user) //Good morning, Mr. Grace.
+	if(awakened)
+		return
+	awakened = TRUE
+	user.visible_message("<span class='boldwarning'>[src] begins to rattle. He thirsts.</span>", "<span class='his_grace'>You flick [src]'s latch up. You hope this is a good idea.</span>")
+	name = "His Grace"
+	desc = "A bloodthirsty artifact created by a profane rite."
+	gender = MALE
+	adjust_bloodthirst(1)
+	force_bonus = HIS_GRACE_FORCE_BONUS * length(contents)
+	playsound(user, 'sound/effects/pope_entry.ogg', 100)
+	move_gracefully()
+
+/obj/item/his_grace/proc/move_gracefully()
+	SIGNAL_HANDLER
+
+	if(!awakened)
+		return
+	var/static/list/transforms
+	if(!transforms)
+		var/matrix/M1 = matrix()
+		var/matrix/M2 = matrix()
+		var/matrix/M3 = matrix()
+		var/matrix/M4 = matrix()
+		M1.Translate(-1, 0)
+		M2.Translate(0, 1)
+		M3.Translate(1, 0)
+		M4.Translate(0, -1)
+		transforms = list(M1, M2, M3, M4)
+
+	animate(src, transform = transforms[1], time = 0.2, loop = -1)
+	animate(transform = transforms[2], time = 0.1)
+	animate(transform = transforms[3], time = 0.2)
+	animate(transform = transforms[4], time = 0.3)
+
+/obj/item/his_grace/proc/drowse() //Good night, Mr. Grace.
+	if(!awakened || ascended)
+		return
+	var/turf/T = get_turf(src)
+	T.visible_message("<span class='boldwarning'>[src] slowly stops rattling and falls still, His latch snapping shut.</span>")
+	playsound(loc, 'sound/weapons/batonextend.ogg', 100, TRUE)
+	name = initial(name)
+	desc = initial(desc)
+	icon_state = initial(icon_state)
+	animate(src, transform = matrix())
+	gender = initial(gender)
+	force = initial(force)
+	force_bonus = initial(force_bonus)
+	awakened = FALSE
+	bloodthirst = 0
+
+/obj/item/his_grace/proc/consume(mob/living/meal) //Here's your dinner, Mr. Grace.
+	if(!meal)
+		return
+	var/victims = 0
+	meal.visible_message("<span class='warning'>[src] swings open and devours [meal]!</span>", "<span class='his_grace big bold'>[src] consumes you!</span>")
+	meal.adjustBruteLoss(200)
+	playsound(meal, 'sound/misc/desceration-02.ogg', 75, TRUE)
+	playsound(src, 'sound/items/eatfood.ogg', 100, TRUE)
+	meal.forceMove(src)
+	force_bonus += HIS_GRACE_FORCE_BONUS
+	prev_bloodthirst = bloodthirst
+	if(prev_bloodthirst < HIS_GRACE_CONSUME_OWNER)
+		bloodthirst = max(length(contents), 1) //Never fully sated, and His hunger will only grow.
+	else
+		bloodthirst = HIS_GRACE_CONSUME_OWNER
+	for(var/mob/living/C in contents)
+		if(C.mind)
+			victims++
+	if(victims >= victims_needed)
```

Comment:
Seems a bit odd that this is recalculated every time He eats someone rather than just tracked as a variable on Him.